### PR TITLE
[WIP] add instance normalization

### PR DESCRIPTION
**Pull request by crcrpar**
Accessing #5229 

This implements Instance Normalization, but this is a draft atm (20180910).

# TODO
- [ ] Backprop
- [ ] Test
- [ ] Document

Any comments are welcome!

### CODE REVIEW

**Comment by crcrpar** [comment]
Hi,

To use cuDNN BatchNormalization, gamma and beta have to be 1-dimensional.
Is this correct?

**Comment by crcrpar** [comment]
Since this is a bit old and I changed the design, close this and submit a new PR.
